core/src/Kokkos_CudaSpace.hpp

```diff
@@ -597,8 +597,26 @@ class SharedAllocationRecord<Kokkos::CudaSpace, void>
       const ExecutionSpace& /*exec_space*/, const Kokkos::CudaSpace& arg_space,
       const std::string& arg_label, const size_t arg_alloc_size,
       const RecordBase::function_type arg_dealloc = &base_t::deallocate)
-      : SharedAllocationRecord(arg_space, arg_label, arg_alloc_size,
-                               arg_dealloc) {}
+      : base_t(
+#ifdef KOKKOS_ENABLE_DEBUG
+            &SharedAllocationRecord<Kokkos::CudaSpace, void>::s_root_record,
+#endif
+            Impl::checked_allocation_with_header(arg_exec_space, arg_space,
+                                                 arg_label, arg_alloc_size),
+            sizeof(SharedAllocationHeader) + arg_alloc_size, arg_dealloc,
+            arg_label),
+        m_tex_obj(0),
+        m_space(arg_space) {
+
+    SharedAllocationHeader header;
+
+    this->base_t::_fill_host_accessible_header_info(header, arg_label);
+
+    // Copy to device memory
+    Kokkos::Impl::DeepCopy<CudaSpace, HostSpace>(
+        arg_exec_space, RecordBase::m_alloc_ptr, &header,
+        sizeof(SharedAllocationHeader));
```

Comment:
This overload should use the synchronous version, right?
```suggestion
          Impl::checked_allocation_with_header(arg_space, arg_label,
                                               arg_alloc_size),
          sizeof(SharedAllocationHeader) + arg_alloc_size, arg_dealloc,
          arg_label),
      m_tex_obj(0),
      m_space(arg_space) {

  SharedAllocationHeader header;

  this->base_t::_fill_host_accessible_header_info(header, arg_label);

  // Copy to device memory
  Kokkos::Cuda exec;
  Kokkos::Impl::DeepCopy<CudaSpace, HostSpace>(
      exec, RecordBase::m_alloc_ptr, &header, sizeof(SharedAllocationHeader));
  exec.fence(
      "SharedAllocationRecord<Kokkos::CudaSpace, "
      "void>::SharedAllocationRecord(): fence after copying header from "
      "HostSpace");
```

Reply:
Did I make a mistake? I think I just copied over what was getting called?